apps/server/src/modules/deletion/entity/deletion-request.entity.ts

```diff
@@ -0,0 +1,60 @@
+import { Entity, Index, Property } from '@mikro-orm/core';
+import { BaseEntityWithTimestamps } from '@shared/domain/entity/base.entity';
+import { ObjectId } from '@mikro-orm/mongodb';
+import { EntityId } from '@shared/domain';
+import { DeletionDomainModel } from '../domain/types/deletion-domain-model.enum';
+import { DeletionStatusModel } from '../domain/types/deletion-status-model.enum';
+
+export interface DeletionRequestEntityProps {
+	id?: EntityId;
+	domain: DeletionDomainModel;
+	deleteAfter: Date;
+	itemId: EntityId;
+	status: DeletionStatusModel;
+	createdAt?: Date;
+	updatedAt?: Date;
+}
+
+@Entity({ tableName: 'deletionrequests' })
+@Index({ properties: ['_itemId', 'domain'] })
+export class DeletionRequestEntity extends BaseEntityWithTimestamps {
+	@Property()
+	deleteAfter: Date;
+
+	@Property()
+	_itemId: ObjectId;
+
+	get itemId(): EntityId {
+		return this._itemId.toHexString();
+	}
```

Comment:
Please change the data type here to just a plain `string`/`EntityId` (not `ObjectId`) - right now (in case of deleting users) we'll put only the ids in the ObjectId format (as this is the current format of the users' ids), but as the structure of this collection is generic (in a sense that it can be re-used for any other domain), I think we should use just a plain `string` as a data type here to make sure we can store any `string`-typed id here, not just an `ObjectId`. Thanks to this we'll never need to run a migration on this field (unless we'll use an integer as an id in any domain, but I think this is very unlikely and we can always store these ints formatted as strings as well if needed). The `string` type here is the most useful to have here in this field in my opinion, generally speaking.

Reply:
Also, I would propose to change the name of the field to e.g. `targetRefId` to make it more obvious that this is the reference to the object that's the subject of the performed deletion (I have a feeling that calling it an _item_ can be not that intuitive for someone else).

Reply:
done